### PR TITLE
Fixes lightbulb injections

### LINKED ISSUE #55071: Plasma in Lightbulb can be injected endlessly
Injecting Plasma into lightbulbs behaves weirdly. You can inject an infinite amount, while the explosion by itself seems to be capped at 5 or 10 units of Plasma. It still explodes normally, just seems weaker than it should, but I understand why this would be coded to have a very low maxcap.

Other things that accept plasma injection have a max you can inject, while lightbulbs seem infinite. I tried with 400u on a private server and the explosion still seemed the same.

When injecting, the syringe injects all it's contents, instead of asking you to click 3 times if the syringe is set to inject only 5u. It gets suck on "Inject", forcing you to activate it in hand to change it's mode manually, when it's normal operation is to change as soon as it's empties itself.

## Round ID:

N/A

## Testmerges:

N/A

## Reproduction:

1: Grind Plasma
2: Fill a syringe with Plasma
3: Grab a lightbulb from the socket
4: Inject the lightbulb
5: Repeat endlesly

### PR DESCRIPTION
**Pull request by CoffeeDragon16**
## About The Pull Request

Fixes #55071. Light bulb injections worked odd, and I simply made it work the same as it does for energy cells. This makes it so you can only inject the plasma once, as intended.

## Why It's Good For The Game

Fixes are good, and bugs are bad.

## Changelog
:cl:
fix: You can no longer inject infinite plasma into a light bulb with no addition effect
/:cl:

### CODE REVIEW

File: code/modules/power/lighting.dm [hunk 1]
```diff
@@ -881,21 +882,9 @@
...
-		if(S.reagents.has_reagent(/datum/reagent/toxin/plasma, 5))
-
-			rigged = TRUE
-
-		S.reagents.clear_reagents()
-	else
-		..()
-	return
+/obj/item/light/on_reagent_change(changetype)
+	rigged = (reagents.has_reagent(/datum/reagent/toxin/plasma, 5)) //has_reagent returns the reagent datum
```
Comment (bobbahbrown) [inline code/modules/power/lighting.dm:886]:
```suggestion
	rigged = reagents.has_reagent(/datum/reagent/toxin/plasma, 5) //has_reagent returns the reagent datum
```

you don't really need to wrap this in brackets i don't think

**Comment by itseasytosee** [comment]
The unintended consequences of this PR are that you can now smuggle meth in lightbulbs.

File: code/modules/power/lighting.dm [hunk 2]
```diff
@@ -863,6 +863,7 @@
 
 /obj/item/light/Initialize()
 	. = ..()
+	create_reagents(5, INJECTABLE | DRAINABLE)
```
Comment (Jared-Fogle) [inline code/modules/power/lighting.dm:866]: Magic number--move 5 out to a define and then `#undef` it at the end.

File: code/modules/power/lighting.dm [hunk 3]
```diff
@@ -881,21 +882,9 @@
...
-		if(S.reagents.has_reagent(/datum/reagent/toxin/plasma, 5))
-
-			rigged = TRUE
-
-		S.reagents.clear_reagents()
-	else
-		..()
-	return
+/obj/item/light/on_reagent_change(changetype)
+	rigged = reagents.has_reagent(/datum/reagent/toxin/plasma, 5) //has_reagent returns the reagent datum
```
Comment (Jared-Fogle) [inline code/modules/power/lighting.dm:886]: Then use that define here.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by TheBonded** [comment]
> The unintended consequences of this PR are that you can now smuggle meth in lightbulbs.

does this mean you can fill lightbulbs with other chemicals that react when heated, and have them react when the lightbulb is turned on?

stabilized TaTP lightbulb meta

File: code/modules/power/lighting.dm [hunk 4]
```diff
@@ -881,21 +885,9 @@
...
-
-			rigged = TRUE
-
-		S.reagents.clear_reagents()
-	else
-		..()
-	return
+/obj/item/light/on_reagent_change(changetype)
+	rigged = (reagents.has_reagent(/datum/reagent/toxin/plasma, LIGHT_REAGENT_CAPACITY)) //has_reagent returns the reagent datum
+	return ..()
```
Comment (itseasytosee) [inline code/modules/power/lighting.dm:890]:
```suggestion
	return ..()
	#undef LIGHT_REAGENT_CAPACITY
```

**Comment by CoffeeDragon16** [comment]
sadly thebonded its snowflaked to plasma so that's someone else's job to get working

File: code/modules/power/lighting.dm (hunk 4 above)
[the remark this answers is not in the document]
Comment (CoffeeDragon16) [reply code/modules/power/lighting.dm:890]: thanks i totally forgot to undef

**Comment by TheBonded** [comment]
does injecting it with a bluespace syringe put in 60u instead of 15?

**Comment by TemporalOroboros** [comment]
@TheBonded Injecting it used to just rig it and eat all of the reagents in the syringe. Now it can store up to 5u of reagents and if all of that's plasma it will explode when it's used.

File: code/modules/power/lighting.dm [hunk 5]
```diff
@@ -881,21 +885,11 @@
...
+	
+	#undef LIGHT_REAGENT_CAPACITY
```
Comment (Jared-Fogle) [inline code/modules/power/lighting.dm:892]:
```suggestion
#undef LIGHT_REAGENT_CAPACITY
```

**Pull request merged**